src/Common/Commands.Common.Authentication.Test/TestHttpOperationsFactory.cs

```diff
@@ -0,0 +1,103 @@
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.Commands.Common.Authentication.Test
+{
+    public class TestHttpOperationsFactory : IHttpOperationsFactory
+    {
+        private TestHttpOperationsFactory()
+        {
+        }
+
+        IDictionary<string, object> _responses = new Dictionary<string, object>(StringComparer.OrdinalIgnoreCase);
+        public IHttpOperations<T> GetHttpOperations<T>()
+        {
+            return new TestHttpOperations<T>(_responses);
+        }
+
+        public static IHttpOperationsFactory Create<T>(IDictionary<string, T> responses)
+        {
+            var factory = new TestHttpOperationsFactory();
+            foreach (var response in responses)
+            {
+                factory._responses.Add(response.Key, response.Value);
+            }
+
+            return factory;
+        }
+
+        class TestHttpOperations<T> : IHttpOperations<T>
+        {
+
+            IDictionary<string, T> _responses = new Dictionary<string, T>(StringComparer.OrdinalIgnoreCase);
+
+            public TestHttpOperations(IDictionary<string, T> responses)
+            {
+                foreach (var responsePair in responses)
+                {
+                    _responses.Add(responsePair);
+                }
+            }
+            public TestHttpOperations(IDictionary<string, object> responses)
+            {
+                foreach (var responsePair in responses)
+                {
+                    _responses.Add(responsePair.Key, (T)(responsePair.Value));
+                }
+            }
+
+            public Task DeleteAsync(string requestUri, CancellationToken token)
+            {
+                throw new NotImplementedException();
+            }
+
+            public Task<T> GetAsync(string requestUri, CancellationToken token)
+            {
+                if (!_responses.ContainsKey(requestUri))
+                {
+                    throw new InvalidOperationException(string.Format("Unexpected request Uri '{0}'", requestUri));
+                }
+
+                return Task.FromResult(_responses[requestUri]);
+            }
+
+            public Task<bool> HeadAsync(string requestUri, CancellationToken token)
+            {
+                throw new NotImplementedException();
+            }
+
+            public Task<IEnumerable<T>> ListAsync(string requestUri, CancellationToken token)
+            {
+                throw new NotImplementedException();
+            }
+
+            public Task<T> PutAsync(string requestUri, T payload, CancellationToken token)
+            {
+                throw new NotImplementedException();
+            }
+
+            public IHttpOperations<T> WithHeader(string name, IEnumerable<string> value)
+            {
+                return this;
```

Comment:
Shouldn't this throw?

Reply:
No, because we simply aren't using headers in the test operations factory, but it should allow specifying headers in any code that uses it.

Reply:
👍 ... maybe a comment indicating as much?